spec/queries/credit_notes_query_spec.rb

```diff
@@ -362,6 +362,20 @@
     end
   end
 
+  context "when billing entity id filter applied" do
+    let(:billing_entity) { create(:billing_entity, organization:) }
+    let(:filters) { {billing_entity_id: billing_entity.id} }
+
+    let!(:matching_credit_note) { create(:credit_note, customer:, invoice: create(:invoice, billing_entity:)) }
```

Comment:
we don't use `let!` lets do let a use the `before` block

Reply:
interestingly, in this spec they exactly were used, that's why I copied this approach